tests/python/gaia-ui-tests/gaiatest/apps/system/regions/cards_view.py

```diff
@@ -72,6 +73,9 @@ def close_app(self, app):
         self.wait_for_card_ready(app)
         self.marionette.find_element(*self._close_button_locator(app)).tap()
         self.wait_for_element_not_present(*self._app_card_locator(app))
+        if len(self.cards) == 0:
+            self.marionette.switch_to_frame()
+            self.wait_for_condition(lambda m: self.apps.displayed_app.name == Homescreen.name)
```

Comment:
I still believe we should add this condition in the test. To me, it is part of the test logic and it should be in the test itself, rather than in the app object. We've been using this condition in several other tests, like [test_ftu_skip_tour.py
][link] or [test_ftu_with_tour.py][link]
[link]:https://github.com/mozilla-b2g/gaia/blob/master/tests/python/gaia-ui-tests/gaiatest/tests/functional/ftu/test_ftu_skip_tour.py#L111
[link]:https://github.com/mozilla-b2g/gaia/blob/master/tests/python/gaia-ui-tests/gaiatest/tests/functional/ftu/test_ftu_with_tour.py#L47